pandas/core/indexes/datetimes.py

```diff
@@ -573,12 +572,13 @@ def _generate(cls, start, end, periods, name, freq,
                                               freq=freq, name=name)
                 else:
                     index = _generate_regular_range(start, end, periods, freq)
-
                 if tz is not None and getattr(index, 'tz', None) is None:
-                    index = conversion.tz_localize_to_utc(_ensure_int64(index),
-                                                          tz,
-                                                          ambiguous=ambiguous)
-                    index = index.view(_NS_DTYPE)
+                    arr = conversion.tz_localize_to_utc(_ensure_int64(index),
```

Comment:
``tz_localize_to_utc`` returns an array and not a ``DatetimeIndex`` so I then convert this array to a ``DatetimeIndex`` called ``index`` so I can pass ``index.values`` to ``_simple_new`` below